### ISSUE TITLE
Handle field access with Python 2 unicode literals

### ISSUE THREAD

**Issue by ezyang** [protocolbuffers/protobuf#3658]
I was a bit perplexed by the following error:

```
from __future__ import unicode_literals
import mything_pb
p = mything_pb.MyProto()
p.ClearField("my_field")
```

This doesn't work on Python 2, it will throw:

```
TypeError: field name must be a string
```

This is because `unicode_literals` turns the string literal into a Unicode literal, but Python protobufs is only doing a native string test to decide if the field name is OK or not. It would be great if it could handle both cases. (This code works on Python 3 because Unicode strings are the "native" string.)

To work around this problem, call `future.utils.bytes_to_native_str` on a binary string literal first.

**Comment by pherl** [comment]
Hmm, I cannot reproduce it locally. It works fine on my machine (python 2.7.13). The relative code is at: https://github.com/google/protobuf/blob/fa5a69e73b0dd667ff15062adbc170310d440ee9/python/google/protobuf/internal/python_message.py#L831 The field name map was created during the import. So I guess if you import unicode_literals before the module import, the map key will be a unicode literal? Feel free to send out a PR if there's a simple fix.

**Comment by ex-nerd** [comment]
I just started running into this with some of my own code today.  The odd thing is that things work fine with the python installed by homebrew on my Mac.  But they fail under various flavors of Linux.

**Comment by anandolee** [comment]
Which Python 2? We do not support python2.6
Python2.7 and above should work fine

**Comment by ex-nerd** [comment]
This is python 2.7 for me

**Comment by anandolee** [comment]
This error is fixed in internal around Aug, 2018. And also in master:
https://github.com/protocolbuffers/protobuf/blame/master/python/google/protobuf/pyext/message.cc#L1810

 Expect to see the fix in 3.7.0 release

**Comment by anandolee** [comment]
Closing it as 3.7.0 is already released
